app.rb

```diff
@@ -0,0 +1,220 @@
+    age = pick_age
+    name = pick_name
+    specialization = rip_specialization
+    [age, name, specialization]
+  end
+
+  def create_person
+    print "\nDo you want to create a student(1) or a teacher (2)? [Input the number]: "
+    @client_input = gets.chomp
+    create_person unless client_input_valid?(client_input, %w[1 2])
+
+    if @client_input == '1'
+      age, name, has_parent_permission = student_info
+      person = Student.new(age, @classroom_default, name, parent_permission: has_parent_permission)
+    else
+      age, name, specialization = teacher_info
+      person = Teacher.new(age, specialization, name)
+    end
+
+    @people << person
+    puts 'Person created successfully'
+  end
+
+  def rip_title
+    print 'Title: '
+    title = gets.chomp
+    title.empty? ? rip_title : title
+  end
+
+  def rip_author
+    print 'Author: '
+    author = gets.chomp
+    author.empty? ? rip_author : author
+  end
+
+  def create_book
+    print ' '
+    title = rip_title
+    author = rip_author
+    book = Book.new(title, author)
+    @books << book
+    puts 'Book Created successfully'
+  end
+
+  def list_books
+    @books.each_with_index { |book, index| puts "#{index}) Title: \"#{book.title}\", Author: \"#{book.author}\"" }
+  end
+
+  def list_persons
+    @people.each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def rip_specified_book
+    puts "\nSelect a book from the following list by number"
+    list_books
+    specified_book_index = gets.chomp
+    (0..@books.length).include?(specified_book_index.to_i) ? specified_book_index.to_i : rip_specified_book
+  end
+
+  def rip_specified_person
+    puts "\nSelect a person from the following list by number (not id)"
+    list_persons
+    specified_person_index = gets.chomp
+    (0..@people.length).include?(specified_person_index.to_i) ? specified_person_index.to_i : read_desired_person
+  end
+
+  def rip_specified_date
+    print "\nDate: "
+    gets.chomp
+  end
+
+  def create_rental
+    return print 'Please add a book first' if @books.empty?
+    return print 'Please add a person first' if @people.empty?
+
+    books = @books[rip_specified_book]
+    person = @people[rip_specified_person]
+    date = rip_specified_date
+
+    rental = Rental.new(date, person, books)
+
+    @rental << rental
+    puts 'Rental created successfully'
+  end
+
+  def create_client(client_input)
+    case client_input
+    when '3'
+      create_person
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    end
+  end
+
+  def list_id_rentals
+    return puts 'Please add a rental first' if @rental.empty?
+
+    puts "\nID of person: "
+    # @people.each do |person|
+    #   puts "ID: #{person.id}, [#{person.class}] Name: #{person.name}, Age: #{person.age}"
+    # end
+    person_id = gets.chomp.to_i
+    rentals = @rental.select { |rent| rent.person.id == person_id }
```

Comment:
- [ ] I can successfully create student, teacher, books rentals. But when I want to list all rentals for a given person id, I am getting an error `undefined method id for #<Book:0x000000010320fe28`
<img width="1088" alt="Screenshot 2022-05-20 at 10 47 12" src="https://user-images.githubusercontent.com/77038610/169469496-be215f17-e073-415f-a361-d29c5ea97b1f.png">

Reply:
Thank you @ShoiraTa for pointing that out